### PR TITLE
Regex.Tests: Disable test that tries to blow the stack, on AOT

### PR DESCRIPTION
**Pull request by radical**
And re-enable `System.Text.RegularExpressions.Unit.Tests`.

`TryEnsureSufficientExecutionStack` doesn't work correctly on wasm/aot, which causes a stack overflow.

Issue: [#66118](https://github.com/dotnet/runtime/issues/66118)

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

File: eng/pipelines/common/platform-matrix.yml [hunk 1]
```diff
@@ -28,267 +28,267 @@ parameters:
 jobs:
 
 # Linux arm
-- ${{ if or(containsValue(parameters.platforms, 'Linux_arm'), in(parameters.platformGroup, 'all', 'gcstress')) }}:
```
Comment (vargaz) [inline eng/pipelines/common/platform-matrix.yml:31]: Unrelated changes ?

**Comment by danmoseley** [comment]
Why doesn't it work on WASM?

**Comment by radical** [comment]
> Why doesn't it work on WASM?

https://github.com/dotnet/runtime/issues/66118#issuecomment-1058575815

**Comment by danmoseley** [comment]
@radical are you still planning to merge this? I guess that would allow us to remove 'blocking-clean-ci' from #66118.

**Comment by radical** [comment]
> @radical are you still planning to merge this? I guess that would allow us to remove 'blocking-clean-ci' from #66118.

This needs some more work. The stack is still blowing up. As you figured, it's completely disabled right now, so we can remove the `blocking-clean-ci` label.

**Pull request closed without merging**